src/Connection.php

```diff
@@ -21,9 +21,27 @@
  * @method array errorInfo()
  * @method mixed getAttribute(int $attribute)
  * @method bool inTransaction()
- * @method string lastInsertId(string $name = null)
+ * @method string lastInsertId(?string $name = null)
  * @method string quote(mixed $string, int $parameterType = PDO::PARAM_STR)
  * @method mixed setAttribute(int $attribute, mixed $value)
+ *
+ * @phpstan-type dsnArgs array{
```

Comment:
For non-classlike types, I prefer to see lower-snake-case, with the type as a suffix if possible. That helps to distinguish them from classlike types, and follows the PHPStan naming style pretty closely (though we cannot use dashes in userland custom types.) So, `dsn_args_array` or something like that.